### DISCUSSION TITLE
HoneyGain AutoPot + Earnapp Container not starting on Rasberry Pi 4 Raspbian + Docker

### DISCUSSION THREAD

**Discussion by ByQwank** [engageub/InternetIncome#148] [General]
HoneyGain AutoPot + Earnapp Container not starting on Rasberry Pi 4 Raspbian + Docker
Normal HoneyGain, and my IpPawns container work - The Autopot for HoneyGain and Earnapp Container do not.

Using Test Branch, Direct Connection off - No proxies on this one.

**Comment by engageub** [comment]
Hello,
Honeygain pot which is recently added supports only arm64 and amd64 architectures.
Are you using armv7l architecture?
To check your CPU architecture, please run the following command.
```
uname -m 
```

Please find the supported architectures for honeygain pot.
https://hub.docker.com/r/xterna/honeygain-pot/tags

Regarding earnapp, did you see any errors in the console while running. 
If you are using direct connection on raspberry pi, you can also download the app directly from earnapp website.
However would like to know what errors you see while starting it. 
Could you please enable logs in properties file and see if there are any errors. 

If there are any issues related to permissions, you may have to add  `--privileged` option while starting earnapp. 
But it gives special access to container which raises security concerns. 


What `--privileged` Does
1. Access to All Devices: The container gets access to all the devices on the host, not just those explicitly specified with --device flags. This means it can interact with hardware devices directly.

2. Extended Linux Capabilities: It gives the container all the capabilities of the host system, as opposed to the default set of capabilities that Docker containers have. This includes capabilities like SYS_ADMIN, SYS_MODULE, SYS_RAWIO, and more, which are generally restricted.

3. Mounting File Systems: The container can mount file systems from the host. This means it can use mount commands to interact with the host’s file systems, which can be a security risk.

4. Manipulate Kernel Parameters: The container can interact with and change kernel parameters, which might affect the host system's behavior.

Thank you

**Comment by engageub** [comment]
An Issue has been raised to support ARMV7 architecture for honeygain pot. Please find the link below. 

https://github.com/XternA/honeygain-pot/issues/3

Thank you

**Comment by engageub** [comment]
Hi,
Could you please update the earnapp logs here.

Thank you

**Comment by engageub** [comment]
Hello,
The test branch has been updated to include a new option: `USE_EARNAPP_PRIVILEGED`. On Raspberry Pi devices, which typically lack a real-time clock (RTC), Earnapp requires accurate time information when initializing the container. The absence of an RTC often results in time-related errors in the logs. By utilizing the `--privileged` option, the container operates normally, bypassing these issues.

You may either install earnapp directly or set `USE_EARNAPP_PRIVILEGED` to `true` in `properties.conf` . However, it is not recommended to set this option to true due to security reasons. 


Thank you
